### PR TITLE
Added extra test for preempt with checkpoint

### PR DESCRIPTION
**Pull request by tpai26**
#### Bug/feature Description
Adding a test so check that if job is preempted with checkpoint, requeue or suspension it will not show the message "Job will never run"

#### Testing logs/output
[pbs_preemption_all_3_logs.txt](https://github.com/PBSPro/pbspro/files/2288185/pbs_preemption_all_3_logs.txt)

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [x] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

**Comment by tpai26** [comment]
@agrawalravi90 I have addressed your comments and added the new log files

File: test/tests/functional/pbs_preemption.py [hunk 1]
```diff
@@ -81,8 +84,99 @@ def test_preempted_never_run(self):
...
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom.shortname,
+                            expect=True)
+
+        # Create checkpoint
+        chk_script = """#!/bin/bash
+kill $1
+exit 0
+"""
+        self.chk_file = self.du.create_temp_file(body=chk_script)
+        self.du.chmod(path=self.chk_file, mode=0o755)
```
[the remark this answers is not in the document]
Comment (tpai26) [reply test/tests/functional/pbs_preemption.py]: I guess this happened when I did autopep8, and the test ran just fine so I never caught it :) I have fixed it .

File: test/tests/functional/pbs_preemption.py [hunk 2]
```diff
@@ -42,47 +42,93 @@ class TestPreemption(TestFunctional):
...
+        self.attrs = {ATTR_l + '.select': '1:ncpus=1'}
+
+        # preempt jobs and check logs
+        self.submit_and_preempt_jobs(preempt_order='C')
+
+        # remove the checkpoint file
+        try:
+            os.remove(self.chk_file)
+        except OSError:
+            pass
```
[the remark this answers is not in the document]
Comment (tpai26) [reply test/tests/functional/pbs_preemption.py]: I have removed this try catch. We do not need to delete a temp file created with default values in /tmp folder. The PTLTestRunner will delete it.

File: test/tests/functional/pbs_preemption.py [hunk 3]
```diff
@@ -42,63 +42,68 @@ class TestPreemption(TestFunctional):
...
+            preempted_by = 'checkpointing'
+        elif preempt_order == 'S':
+            job_state = 'S'
+            preempted_by = 'suspension'
 
-        # Set ncpus to 2
-        attr = {'resources_available.ncpus': '2'}
-        self.server.manager(MGR_CMD_SET, NODE, attr, self.mom.shortname,
-                            expect=True)
+        # set preempt order to suspension
```
[the remark this answers is not in the document]
Comment (tpai26) [reply test/tests/functional/pbs_preemption.py]: done

File: test/tests/functional/pbs_preemption.py [hunk 4]
```diff
@@ -120,12 +120,6 @@ def test_never_run_preempt_checkpoint(self):
         # preempt jobs and check logs
         self.submit_and_preempt_jobs(preempt_order='C')
 
-        # remove the checkpoint file
```
Comment (agrawalravi90) [inline test/tests/functional/pbs_preemption.py]: Why did you remove this ?

File: test/tests/functional/pbs_preemption.py [hunk 5]
```diff
@@ -81,8 +84,99 @@ def test_preempted_never_run(self):
...
+        self.scheduler.set_sched_config({'preempt_order': 'C'})
+
+        # Create high priority queue
+        a = {'queue_type': 'execution',
+             'started': 'True',
+             'enabled': 'True',
+             'Priority': 200}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, "expressq")
+
+        # submit a job to regular workq
```
[the remark this answers is not in the document]
Comment (tpai26) [reply test/tests/functional/pbs_preemption.py]: The priority of workq is lower than the new added expressq. New low priority queue is not added because that would be having an extra low priority queue in the setup. Also if there is some problem with the workq that would mean there is a problem with setup of PTL which this kind of test might catch.

File: test/tests/functional/pbs_preemption.py (hunk 4 above)
Comment (tpai26) [reply test/tests/functional/pbs_preemption.py]: The PTLTestRunner deletes it.

**Comment by agrawalravi90** [comment]
I'm happy with your changes Tejas, thanks for making all the changes.

**Comment by tpai26** [comment]
Thanks for the sign-offs. I have rebased and pushed

**Comment by tpai26** [comment]
Also added final logs

**Comment by agrawalravi90** [comment]
Thanks Tejas, I'll merge this in
